MetaU Capstone Project/Models/MusicSession.m

```diff
@@ -120,16 +113,22 @@ + (void)removeUserFromSession:(NSString *)sessionCode user: ( PFUser * )user wit
         if (session.count > 0) {
             NSMutableArray *users = [session[0] valueForKey:@"activeUsers"];
             
-            
             if (users.count > 1) {
-                // TODO: If user was host, assign new host
-                NSLog(@"[%lu] %@", (unsigned long)users.count, users);
                 
-                NSInteger *count = @(0);
-
-                [users removeObjectIdenticalTo:user.objectId];
-                NSLog(@"[%lu] %@", (unsigned long)users.count, users);
-                [session setValue:users forKey:@"activeUsers"];
+                for (NSInteger i = 0; i < users.count; i++) {
+                    NSString *userObjectID = [users[i] valueForKey:@"objectId"];
+                    if ([user.objectId isEqualToString:userObjectID]) {
+                        [users removeObjectAtIndex:i];
+                    }
+                }
+                
+                NSString *host = [[session[0] valueForKey:@"activeUsers"] valueForKey:@"objectId"];
```

Comment:
Remember to safely unwrap dictionaries.

As mentioned before, calling `valueForKey:` on null dictionaries can cause crashes.
I would suggest something like this:

```
if (session.count == 0) {
  // avoid array-out-of-bounds crash. Return early or however you want to handle this
  return;
}
NSDictionary activeUsers = [session[0] valueForKey:@"activeUsers"];
if (!activeUsers) {
  // Return early or however you want to handle this
  return;
}
NSString *host = [activeUsers valueForKey:@"objectId"];
if (!host) {
  // Return early or however you want to handle this
  return;
}

// ...finally do something with the host property
```